datafusion/sql/src/expr/function.rs

```diff
@@ -415,9 +415,11 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
     ) -> Result<WindowFunctionDefinition> {
         // check udaf first
         let udaf = self.context_provider.get_aggregate_meta(name);
-        // Skip first value and last value, since we expect window builtin first/last value not udaf version
+        // Use the builtin window function instead of the user-defined aggregate function
         if udaf.as_ref().is_some_and(|udaf| {
-            udaf.name() != "first_value" && udaf.name() != "last_value"
+            udaf.name() != "first_value"
+                && udaf.name() != "last_value"
+                && udaf.name() != "nth_value"
```

Comment:
Bypass the udaf and uses the builtin window function in case of window aggregations. 

This was a tricky bit for me and it took me a while to make the connection and realize that window aggregations need not be handled in this UDAF.

Reply:
We may want to replace with UDAF when we convert builtin-window to UDWF #8709

Reply:
Got it :+1: